src/Features/Core/Portable/DocumentHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -14,13 +14,48 @@
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.Implementation.ReferenceHighlighting
+namespace Microsoft.CodeAnalysis.DocumentHighlighting
 {
-    internal abstract class AbstractDocumentHighlightsService : IDocumentHighlightsService
+    internal abstract partial class AbstractDocumentHighlightsService : IDocumentHighlightsService
     {
         public async Task<ImmutableArray<DocumentHighlights>> GetDocumentHighlightsAsync(
             Document document, int position, IImmutableSet<Document> documentsToSearch, CancellationToken cancellationToken)
         {
+            var (succeeded, highlights) = await GetDocumentHighlightsInRemoteProcessAsync(
```

Comment:
I have verified that that is the case.  Trying ot get a session when you're already remote insta-fails as there is no remote-host service interface to even acquire.